### PR TITLE
Added ability to configure accelerometer

### PR DESCRIPTION
**Pull request by dherrada**
I can do this with the other sensors as well if you'd like me to.

### CODE REVIEW

**Comment by dherrada** [comment]
Something odd is going on with github actions. Tested it on this PR, another PR on this library, and a PR on a different circuitpython library. I'll take a look at it.

**Comment by caternuson** [comment]
CircuitPython libraries favor using properties instead of getter/setter functions, see here:
https://circuitpython.readthedocs.io/en/5.3.x/docs/design_guide.html#getters-setters

So you are essentially wanting to add 3 properties:
* `rng` (or maybe `acc_range` or `g-range`?)
* `bandwidth`
* `acc_mode`

There are already several properties defined in this library you can use as a reference.

Also, be careful about Page 0 vs. Page 1 registers. You want to make sure you are accessing register 0x08 from Page 1. I'm not seeing anything that currently access anything in Page 1. But there is at least a `_PAGE_REGISTER` const already defined. And a few places where it is forced to Page 0.

**Comment by dherrada** [comment]
@caternuson Thanks for taking a look at this. I'll make those changes now.

**Comment by dherrada** [comment]
@caternuson Do you think it'd be worth adding the configuration for the gyro and magnetometer? It should be pretty much just copying and pasting the current code I've got (I've made the changes you asked, just have to test).

**Comment by caternuson** [comment]
Sure. Might as well. To keep things clear as to what sensor a parameter is related to, how about these naming prefixes?

* `accel_`
* `gyro_`
* `magnet_`

ex: `accel_range`, `accel_bandwidth`, `accel_mode`, etc.

**Comment by dherrada** [comment]
@caternuson Yeah, I think that makes sense

**Comment by dherrada** [comment]
@caternuson I have tested this and it does work well. The only issue arises when you're trying to use more than one non-fusion mode in the same run since the registers can't be written to even after the mode is switched. I seem to recall encountering this a few months ago, and I thought the issue was on the side of the bno055, but I could be completely misremembering.

**Comment by caternuson** [comment]
Your getters change to Page 1 then set back to Page 0. But none of the setters do that. Is that OK?

**Comment by dherrada** [comment]
> Your getters change to Page 1 then set back to Page 0. But none of the setters do that. Is that OK?
I _think_ so. That being said, I'd feel better about the chances this could cause an issue later on if I had it change back to page 0, so I'll do that.

**Comment by caternuson** [comment]
Ah - just noticed the return type is string, because of use of `bin()`. It's better to have things be symmetric. That is, the getter and setter should be of the same type. For example, this should work:
```python
Adafruit CircuitPython 5.3.0 on 2020-04-29; Adafruit ItsyBitsy M4 Express with samd51g19
>>> import board
>>> import adafruit_bno055
>>> bno = adafruit_bno055.BNO055_I2C(board.I2C())
>>> bno.accel_range
'0b1'
>>> range = bno.accel_range
>>> bno.accel_range = range
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "adafruit_bno055.py", line 407, in accel_range
TypeError: can't convert 'int' object to str implicitly
>>> 
```

It also allows for comparisons, ex:
```python
>>> adafruit_bno055.ACCEL_4G
1
>>> bno.accel_range
'0b1'
>>> bno.accel_range == adafruit_bno055.ACCEL_4G
False
>>> 
```
That should have been `True`.

**Comment by dherrada** [comment]
@caternuson Ah, yeah. That definitely makes sense. I was wanting to display it 'visually' in binary since it is a lot easier to see what's going on that way, but it's definitely a lot better for it to return an int.

File: adafruit_bno055.py [hunk 1]
```diff
@@ -327,6 +389,184 @@ def gravity(self):
...
+        if self.mode in [0x08, 0x09, 0x0A, 0x0B, 0x0C]:
+            raise RuntimeError("Mode must not be a fusion mode")
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_MAGNET_CONFIG_REGISTER)
+        masked_value = 0b01100111 & value
+        self._write_register(_MAGNET_CONFIG_REGISTER, masked_value | mode)
+        self._write_register(_PAGE_REGISTER, 0x00)
+
+    @property
+    def magnet_power_mode(self):
```
Comment (caternuson) [inline adafruit_bno055.py:551]: How about naming this one just `magnet_mode` to better match `accel_mode` and `gyro_mode`? It looks like all three of those are functionally the same thing for each sensor.

And then leave `magnet_operation_mode` as is. That one looks unique to the magnet.

**Comment by caternuson** [comment]
One more suggestion added. But those last updates are working better:
```python
Adafruit CircuitPython 5.3.0 on 2020-04-29; Adafruit ItsyBitsy M4 Express with samd51g19
>>> import board
>>> import adafruit_bno055
>>> bno = adafruit_bno055.BNO055_I2C(board.I2C())
>>> range = bno.accel_range
>>> bno.accel_range = range
>>> bno.accel_range == adafruit_bno055.ACCEL_4G
True
>>> 
```

File: adafruit_bno055.py (hunk 1 above)
Comment (dherrada) [reply adafruit_bno055.py:551]: Yeah, that makes sense

**Comment by caternuson** [comment]
Cool. OK, calling it good at this point. Thanks for all the updates.

**Pull request merged**